### DISCUSSION TITLE
Install of using aws ami fails

### DISCUSSION THREAD

**Discussion by saixarjun** [Security-Onion-Solutions/securityonion#11158] [2.4]
### Version

2.4.0

### Installation Method

Network installation on Red Hat derivative like Oracle, Rocky, Alma, etc.

### Description

installation

### Installation Type

Eval

### Location

cloud

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I tried installing 2.4 with the AMI in AWS. The logs show no clue as to why it fails. I tried both Eval and standard and none worked for me. Please let me know if there are any known issues already around it. I could not see any salt failures and everything was seeming smooth for its 30+ mins of install journey. 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by dougburks** [comment]
Did you follow the documentation at https://docs.securityonion.net/en/2.4/cloud-amazon.html?

What instance type did you use and what are its specs?

**Comment by saixarjun** [comment]
Yes we followed line by line and are using t3a.2xlarge as recommended.

Update from yesterday, We got the installation working but suricata is not
coming up and zeek keeps restarting.
Here is a status output I see :

***@***.*** ~]$ sudo so-status

                             Security Onion Status
                         Container │ Status  │
Details
───────────────────────────────────┼─────────┼─────────────────────────────────
                        so-curator │ running │                     Up 22
hours
                 so-dockerregistry │ running │                     Up 22
hours
                     so-elastalert │ running │                     Up 22
hours
                  so-elastic-fleet │ running │                     Up 22
hours
 so-elastic-fleet-package-registry │ running │           Up 22 hours
(healthy)
                  so-elasticsearch │ running │                     Up 22
hours
                       so-idstools │ running │                     Up 22
hours
                       so-influxdb │ running │           Up 22 hours
(healthy)
                         so-kibana │ running │                     Up 22
hours
                         so-kratos │ running │                     Up 22
hours
                       so-logstash │ running │                     Up 22
hours
                          so-mysql │ running │           Up 22 hours
(healthy)
                          so-nginx │ running │           Up 22 hours
(healthy)
                       so-playbook │ running │                     Up 22
hours
                          so-redis │ running │                     Up 22
hours
                      so-sensoroni │ running │                     Up 22
hours
                            so-soc │ running │                     Up 22
hours
                       so-soctopus │ running │                     Up 22
hours
                          so-steno │ running │                     Up 22
hours
                so-strelka-backend │ running │                     Up 22
hours
            so-strelka-coordinator │ running │                     Up 22
hours
             so-strelka-filestream │ running │                     Up 22
hours
               so-strelka-frontend │ running │                     Up 22
hours
             so-strelka-gatekeeper │ running │                     Up 22
hours
                so-strelka-manager │ running │                     Up 22
hours
                       *so-suricata │ missing │*
                       so-telegraf │ running │                     Up 22
hours
                          * so-zeek │ running │ Up 6 minutes (health:
starting)*

 ❗ Check container logs and /opt/so/log for more details.

I am not sure whether to proceed with this and if there are fixes here.
Please suggest what should I be doing to get this working completely fine.
This is installed as "Standard install".

Regards,
Arjun

**Comment by dougburks** [comment]
What kind of architecture are you trying to build?
https://docs.securityonion.net/en/2.4/architecture.html

Depending on your goals, Eval or Standalone may not be the right kind of installation. You may want a distributed deployment:
https://docs.securityonion.net/en/2.4/architecture.html#distributed
https://docs.securityonion.net/en/2.4/cloud-amazon.html#distributed-grid

**Comment by saixarjun** [comment]
Thank you for providing those details. We are doing a proof of concept
before using the distributed deployment.

Is there a readymade standalone/eval setup we could use to do the proof of
concept?

Regards,
Arjun

On Sat, Aug 26, 2023 at 12:06 AM Doug Burks ***@***.***>
wrote:

> What kind of architecture are you trying to build?
> https://docs.securityonion.net/en/2.4/architecture.html
>
> Depending on your goals, Eval or Standalone may not be the right kind of
> installation. You may want a distributed deployment:
> https://docs.securityonion.net/en/2.4/architecture.html#distributed
> https://docs.securityonion.net/en/2.4/cloud-amazon.html#distributed-grid
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/11158#discussioncomment-6825839>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A36YR7FIVIWNVH77X77ILGDXXDWCBANCNFSM6AAAAAA34SAHSQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <Security-Onion-Solutions/securityonion/repo-discussions/11158/comments/6825839
> @github.com>
>


--

**Comment by dougburks** [comment]
What exactly are you hoping to accomplish in your proof of concept? If you're not sniffing network traffic in the cloud environment, then you don't need to worry about Suricata and Zeek and you could try a simple ManagerSearch installation:
https://docs.securityonion.net/en/2.4/architecture.html#manager-search

**Comment by saixarjun** [comment]
Yeah we wanted to sniff our on-prem enterprise while having just the
sensors on-prem and the manager on cloud.
Let me check the doc you shared and see if that helps. This needs to be
On-Prem, I believe.

On Mon, Aug 28, 2023 at 5:30 PM Doug Burks ***@***.***> wrote:

> What exactly are you hoping to accomplish in your proof of concept? If
> you're not sniffing network traffic in the cloud environment, then you
> don't need to worry about Suricata and Zeek and you could try a simple
> ManagerSearch installation:
> https://docs.securityonion.net/en/2.4/architecture.html#manager-search
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/11158#discussioncomment-6841671>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A36YR7BVWKWVKGVU73PMOL3XXSB7VANCNFSM6AAAAAA34SAHSQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <Security-Onion-Solutions/securityonion/repo-discussions/11158/comments/6841671
> @github.com>
>


--

**Comment by dougburks** [comment]
If the goal of the proof of concept is to sniff traffic in your on-prem enterprise, then you have a few different options:
- perform a Standalone installation in the on-prem network
- install a ManagerSearch in the on-prem network, install a separate forward node in the on-prem network, and then join it to the ManagerSearch
- install a ManagerSearch in the cloud, install a forward node in the on-prem network, and then join it to the ManagerSearch

For more information, please see:
https://docs.securityonion.net/en/2.4/architecture.html
https://docs.securityonion.net/en/2.4/hardware.html

**Comment by saixarjun** [comment]
I could try the 1st and 2nd but the intent won't be fulfilled and I can not
try the 3rd option as the install fails in AWS.

I would appreciate it if you could get the AMI provided in AWS marketplace
tested once or provide a video recording reference where it went successful
on AWS.

Thank you so much for patiently answering my questions.



On Wed, Aug 30, 2023 at 2:09 AM Doug Burks ***@***.***> wrote:

> If the goal of the proof of concept is to sniff traffic in your on-prem
> enterprise, then you have a few different options:
>
>    - perform a Standalone installation in the on-prem network
>    - install a ManagerSearch in the on-prem network, install a separate
>    forward node in the on-prem network, and then join it to the ManagerSearch
>    - install a ManagerSearch in the cloud, install a forward node in the
>    on-prem network, and then join it to the ManagerSearch
>
> For more information, please see:
> https://docs.securityonion.net/en/2.4/architecture.html
> https://docs.securityonion.net/en/2.4/hardware.html
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/11158#discussioncomment-6857382>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A36YR7G7YYXKNI5HE2EZSPLXXZHPNANCNFSM6AAAAAA34SAHSQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <Security-Onion-Solutions/securityonion/repo-discussions/11158/comments/6857382
> @github.com>
>


--

**Comment by dougburks** [comment]
> I would appreciate it if you could get the AMI provided in AWS marketplace tested once

We've tested it many times and it works for us. 

Based on the details you've shared above, the most likely reason for your previous issues is choosing Eval or Standalone. Both of those options enable sniffing which is handled differently in the cloud and is not a goal of your proof of concept. 

Instead of choosing Eval or Standalone, have you tried a ManagerSearch installation as I mentioned above?
https://docs.securityonion.net/en/2.4/architecture.html#manager-search
